### DISCUSSION TITLE
How to get the path of the config that was given?

### DISCUSSION THREAD

**Discussion by msis** [facebookresearch/hydra#2750] [Q&A] [answered]
In this small example, how to get the path of the config that was passed?
```py
from omegaconf import DictConfig, OmegaConf
import hydra

@hydra.main(version_base=None, config_path=".", config_name="config")
def my_app(cfg):
    print(OmegaConf.to_yaml(cfg))
    # how to get the config_name and config_path here???

if __name__ == "__main__":
    my_app()
```

**Comment by odelalleau** [comment, marked as answer]
Not sure if there's a better way but I think you can do (not fully tested):
```python
from hydra.core.hydra_config import HydraConfig

cfg = HydraConfig.get()
config_name = cfg.job.config_name
config_path = [path["path"] for path in cfg.runtime.config_sources if path["schema"] == "file"][0]
```

**Reply by msis** [reply]
I will use this for now.
thanks!

**Comment by francotheengineer** [comment]
This would be a really good attribute to have access to.  Sometimes we just want to make a direct copy of all the files in a config dir for future reference. If the code does not know this location, the files cannot be access. 
This is a good work around:
```
    hydra_config = HydraConfig().get()
    config_sources = hydra_config.runtime.config_sources
    config_sources_provider_paths = {i['provider']:i['path'] for i in config_sources}
```
when `provider` is `command-line` , that's the hydra CLI override, when `provider` is `main` that's the default from the `@hydra.main(..)` decorator.
